bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -134,15 +142,20 @@ public GarbageCollectorThread(ServerConfiguration conf,
         // compaction parameters
         minorCompactionThreshold = conf.getMinorCompactionThreshold();
         minorCompactionInterval = conf.getMinorCompactionInterval() * SECOND;
-        majorCompactionThreshold = conf.getMajorCompactionThreshold();
+        originalMajorCompactionThreshold = majorCompactionThreshold = conf.getMajorCompactionThreshold();
         majorCompactionInterval = conf.getMajorCompactionInterval() * SECOND;
+        medianMajorCompactionThreshold = conf.getMedianMajorCompactionThreshold();
```

Comment:
do we need to set any constraints between `MajorCompactionInterval` `MedianMajorCompactionInterval` and `HighMajorCompactionInterval`? E.g `MajorCompactionInterval` < `MedianMajorCompactionInterval` < `HighMajorCompactionInterval`?

Reply:
that's right